addons/mail/models/mail_thread.py

```diff
@@ -808,6 +809,109 @@ def _routing_reset_bounce(self, email_message, message_dict):
+        if primary_email and primary_email in self._fields:
+            return primary_email
+
+    @api.model
+    def _routing_detect_loop_from_records_domain(self, email_from_normalized):
+        """Return the domain to be used to detect duplicated records created by alias.
+
+        :param email_from_normalized: FROM of the incoming email, normalized
+        """
+        primary_email = self._check_primary_email()
+        if primary_email:
+            return [(primary_email, 'ilike', email_from_normalized)]
+
+        _logger.info('Primary email missing on %s', self._name)
+
+    @api.model
+    def _routing_detect_loop_from_records(self, message_dict, message, routes):
+        """This method returns True if the incoming email should be ignored.
+
+        The goal of this method is to prevent loops which can occur if an auto-replier
+        send emails to Odoo.
+        """
+        email_from = message_dict.get('email_from')
+        if not email_from:
+            return False
+
+        email_from_normalized = tools.email_normalize(email_from)
+
+        # Detect the email address sent to many emails
+        get_param = self.env['ir.config_parameter'].sudo().get_param
+        INCOMING_LIMIT_PERIOD = int(get_param('mail.incoming.limit.period', 120))
+        INCOMING_LIMIT_ALIAS = int(get_param('mail.incoming.limit.alias', 20))
+
+        create_date_limit = datetime.datetime.now() - datetime.timedelta(minutes=INCOMING_LIMIT_PERIOD)
+
+        # Search only once per model
+        models = {
+            self.env[model]
+            for model, thread_id, *__ in routes or []
+            if not thread_id  # Reply to an existing thread
+        }
+
+        for model in models:
+            if not hasattr(model, '_routing_detect_loop_from_records_domain'):
+                continue
+
+            domain = model._routing_detect_loop_from_records_domain(email_from_normalized)
+
+            if not domain:
+                _logger.info('Domain missing for %s, could not find duplicated.', self._name)
+                continue
+
+            mail_incoming_messages_count = model.sudo().search_count(
+                expression.AND([
+                    [('create_date', '>', create_date_limit)],
+                    domain,
+                ]),
+            )
+
+            if mail_incoming_messages_count >= INCOMING_LIMIT_ALIAS:
+                _logger.info('This email address created too many <%s>.', model)
+
+                body = self.env['ir.qweb']._render(
+                    'mail.message_notification_limit_email',
+                    {'email': message_dict.get('to')},
+                    minimal_qcontext=True,
+                    raise_if_not_found=False)
+
+                # Add a reference with a tag, to be able to ignore response to this email
+                references = message_dict.get('message_id', '') + ' ' + tools.generate_tracking_message_id('bounce-email')
```

Comment:
Can't we setup something a bit more hashed / less prone to create conflicts compared to "bounce-email" ?